### ISSUE TITLE
callback: get file: get next chunk: invoke pool: rpcDoRequest: retryUntilAck: retry limit reached after 10 attempts

### ISSUE THREAD

**Issue by facmasterII** [iyear/tdl#95]
callback: get file: get next chunk: invoke pool: rpcDoRequest: retryUntilAck: retry limit reached after 10 attempts

Hello bro, and thanks for the program, is really helpfull

**Comment by facmasterII** [comment]
Wait 5 minutes and put it back on. And I download well. There would be some method for that in the event that this error occurs. Wait a while and try again?

**Comment by facmasterII** [comment]
On a pool with 1000 files, I had to reboot several times. Ideally, it would be automatic, pause and then continue

**Comment by iyear** [comment]
Please see issue #86  for resumable downloads

**Comment by facmasterII** [comment]
Yes, but there would be human intervention. Instead, with my method it simply throws this error. It would be to generate a break and then continue from the point where I left off

**Comment by facmasterII** [comment]
> Please see issue #86 for resumable downloads

My idea is to retry download every 2 minutes. After that it throws the above mentioned error.
Many times it is also due to internet problems and this solution would be great

**Comment by iyear** [comment]
I will improve the mechanism for retries caused by the network.

**Comment by facmasterII** [comment]
> I will improve the mechanism for retries caused by the network.
> 
> But improving your network situation is also a necessary measure, otherwise the download speed will also be very slow.

Wow thanks bro. Yes really needed. I use another app for download media in channel and never get problems in downloads. But dont work in group or saved messages

**Comment by art435** [comment]
I encounter this error too. I think a solution is to remove the retry limit, the tdl should retry until a connection is established.

**Comment by iyear** [comment]
I will change the parameters in the next minor release to minimize the occurrence of errors(but not promise). The RPC_CALL_FAIL error and retries seem to be related to the instability of the Telegram server, which I've also encountered.

**Comment by zengruzhen** [comment]
我也碰到相同问题，不知是否与梯子稳定性有关

**Comment by Bleestones** [comment]
Same error. I would be happy if it could restart automatically. Is there any way to do that?

**Comment by iyear** [comment]
@Bleestones 

`tdl` is not responsible for daemon, you can choose a daemon program for different platforms, for example systemd for Linux.

Command: `tdl dl <OTHER_FLAGS> --continue`

This way tdl will be restarted in case of errors and will continue downloading task sources.
